src/frames.ts

```diff
@@ -566,12 +566,19 @@ export class Frame {
     return this._raceWithCSPError(async () => {
       if (url !== null)
         return (await context.evaluateHandleInternal(addScriptUrl, { url, type })).asElement()!;
+      let result;
       if (path !== null) {
         let contents = await util.promisify(fs.readFile)(path, 'utf8');
         contents += '//# sourceURL=' + path.replace(/\n/g, '');
-        return (await context.evaluateHandleInternal(addScriptContent, { content: contents, type })).asElement()!;
+        result = (await context.evaluateHandleInternal(addScriptContent, { content: contents, type })).asElement()!;
+      } else {
+        result = (await context.evaluateHandleInternal(addScriptContent, { content: content!, type })).asElement()!;
       }
-      return (await context.evaluateHandleInternal(addScriptContent, { content: content!, type })).asElement()!;
+      // Another round trip to the browser to ensure that we receive CSP error messages
+      // (if any) logged asynchronously in a separate task on the content main thread.
+      if (this._page._delegate.cspErrorsAsynchronousForInlineScipts)
```

Comment:
We should probably do the same for inline styles and test it.

Reply:
No, styles are different. I thought the same but the styles seem to be handled asynchronously and they support onload/error events. We already have a [test](https://github.com/microsoft/playwright/blob/master/test/page.spec.js#L725) for this.